### PR TITLE
added support for blacklist

### LINKED ISSUE #1142: Add blacklist to `modules bump-versions`
Add the possibility to not update certain modules using a blacklist for `nf-core modules bump-versions`

### PR DESCRIPTION
**Pull request by KevinMenden**
Added support for a configuration file/ blacklist for the `nf-core modules bump-versions` command.

This is a `.yml` file which can be filled with module names. The module is completely ignored if `False` is given:

e.g.
`star/align: False`

If a version is given, the module is updated/downgrade to that version (if it's different):

`star/align: "some version"`

So you can either set it to `False` or to a specific version.

Fixes #1142 

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/modules/bump_versions.py [hunk 1]
```diff
@@ -120,88 +127,102 @@ def bump_module_version(self, module: NFCoreModule):
             self.failed.append((f"Ignoring mulled container", module.module_name))
             return False
 
-        # Check for correct version and newer versions
+        # Don't update if blocked in blacklist
+        self.bump_versions_config = self.tools_config.get("bump-versions", {})
+        if module.module_name in self.bump_versions_config.keys():
```
Comment (ErikDanielsson) [inline nf_core/modules/bump_versions.py:132]:
```suggestion
        if module.module_name in self.bump_versions_config:
```
Not needed :)

File: nf_core/modules/bump_versions.py [hunk 2]
```diff
@@ -120,88 +127,102 @@ def bump_module_version(self, module: NFCoreModule):
...
+
+        if not config_version:
+            try:
+                response = nf_core.utils.anaconda_package(bp)
+            except LookupError as e:
+                self.failed.append((f"Conda version not specified correctly: {module.main_nf}", module.module_name))
+                return False
+            except ValueError as e:
+                self.failed.append((f"Conda version not specified correctly: {module.main_nf}", module.module_name))
+                return False
```
Comment (ErikDanielsson) [inline nf_core/modules/bump_versions.py:152]:
```suggestion
            except (LookupError, ValueError) as e:
                self.failed.append((f"Conda version not specified correctly: {module.main_nf}", module.module_name))
                return False
            
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -34,6 +34,7 @@
 * Refactored passing of command line arguments to `nf-core` commands and subcommands ([#1139](https://github.com/nf-core/tools/issues/1139), [#1140](https://github.com/nf-core/tools/issues/1140))
 * Check for `modules.json` for entries of modules that are not actually installed in the pipeline [[#1141](https://github.com/nf-core/tools/issues/1141)]
 * Added `<keywords>` argument to `nf-core modules list` for filtering the listed modules. ([#1139](https://github.com/nf-core/tools/issues/1139)
+* Added support for a `bump-versions` configuratoin file [[#1142](https://github.com/nf-core/tools/issues/1142)]
```
Comment (ErikDanielsson) [inline CHANGELOG.md:37]:
```suggestion
* Added support for a `bump-versions` configuration file [[#1142](https://github.com/nf-core/tools/issues/1142)]
```

**Review by ErikDanielsson — APPROVED** [review]
LGTM :+1: Just a few small suggestions

**Comment by KevinMenden** [comment]
Thanks, good suggestions 😁

**Pull request merged**